libs/vkd3d/resource.c

```diff
@@ -297,6 +297,25 @@ static bool vkd3d_sparse_image_may_have_mip_tail(const D3D12_RESOURCE_DESC *desc
             mip_extent.depth < block_extent.depth;
 }
 
+static bool vkd3d_resource_can_be_vrs(struct d3d12_device *device,
+    const D3D12_HEAP_PROPERTIES *heap_properties, const D3D12_RESOURCE_DESC *desc)
+{
+    return device->d3d12_caps.options6.VariableShadingRateTier >= D3D12_VARIABLE_SHADING_RATE_TIER_2 &&
```

Comment:
Need to also check that device actually supports VRS tier 2 in case we used VKD3D_FEATURE_LEVEL=12_2 to override caps.